src/parser.c

```diff
@@ -596,6 +596,17 @@ void read_parameter_list_decl(func_t *func, int anon)
 {
     int vn = 0;
     lex_expect(T_open_bracket);
+
+    char token[MAX_TYPE_LEN];
+    if (lex_peek(T_identifier, token) && !strncmp(token, "void", 4)) {
+        next_token = lex_token();
+        if (lex_accept(T_close_bracket))
```

Comment:
As [C99 specification](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2310.pdf) specifies:

> 6.7.6.3 Function declarators (including prototypes)
> ...
> 10 The special case of an unnamed parameter of type void as the only item in the list specifies that the
function has no parameters.
> ...

This indicates that any forms other than `f(void)` is incorrect, e.g.
```c
f(void n)
f(void, int n)
f(void n, int n)
```

That is, you should expect `T_close_bracket` and rather than any other parameter list identifiers.